File: module_test/src/main/java/com/gas/test/ui/activity/home/mvp/HomePresenter.java
package com.gas.test.ui.activity.home.mvp;


import com.base.lib.di.scope.ActivityScope;
import com.base.lib.mvp.BasePresenter;
import com.gas.test.bean.TestInfoBean;
import com.gas.test.ui.activity.home.HomeAdapter;

import java.util.List;

import javax.inject.Inject;


@ActivityScope
public class HomePresenter extends BasePresenter<HomeContract.Model, HomeContract.View> {


    @Inject
    HomeAdapter mAdapter;

    @Inject
    public HomePresenter(HomeContract.Model model, HomeContract.View rootView) {
        super(model, rootView);
    }

    public void showTestInfos() {
        List<TestInfoBean> infos = mModel.getInfoItems();
        mAdapter.setNewData(infos);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

    }
}

File: module_test/src/main/java/com/gas/test/ui/home/mvp/HomePresenter.kt
package com.gas.test.ui.home.mvp

import com.base.lib.di.scope.ActivityScope
import com.base.lib.mvp.BasePresenter

import javax.inject.Inject

@ActivityScope
class HomePresenter @Inject
constructor(model: HomeContract.Model, rootView: HomeContract.View) : BasePresenter<HomeContract.Model, HomeContract.View>(model, rootView) {


    override fun onDestroy() {

    }


}
